astropy/wcs/wcs.py

```diff
@@ -3068,6 +3068,34 @@ def _as_mpl_axes(self):
         from ..visualization.wcsaxes import WCSAxes
         return WCSAxes, {'wcs': self}
 
+    def contains(self, coord, mode='all'):
+        """
+        Determines if a given SkyCoord is contained in the wcs footprint.
+
+        Parameters
+        ----------
+        coord : `~astropy.coordinates.SkyCoord`
+            The coordinate to check if it is withing the wcs coordinate.
+
+        Returns
+        -------
+        response : bool
+           True means the wsc footprint contains the coordinate, False means it does not.
+        """
+
+        with warnings.catch_warnings():
+            #  Suppress warnings since they just mean we didn't find the coordinate
+            warnings.simplefilter("ignore")
+            try:
+                x,y = coord.to_pixel(self, mode=mode)
```

Comment:
Perhaps you should add the `origin` keyword here?  Or alternatively, replace ``mode=mode`` with just ``**kwargs`` and note in the docstring that the kwargs get passed into `to_pixel`?